src/tests/integration/test_connection.py

```diff
@@ -0,0 +1,135 @@
+            "postgresql",
+        ],
+        [
+            "setup_duckDB_native",
+            DBAPIConnection,
+            "DuckDBPyConnection",
+            "duckdb",
+        ],
+        [
+            "setup_duckDB",
+            SQLAlchemyConnection,
+            "duckdb:////tmp/db-duckdb",
+            "duckdb",
+        ],
+        [
+            "setup_postgreSQL",
+            partial(SQLAlchemyConnection, alias="some-postgres"),
+            "some-postgres",
+            "postgresql",
+        ],
+        [
+            "setup_duckDB_native",
+            partial(DBAPIConnection, alias="some-duckdb"),
+            "some-duckdb",
+            "duckdb",
+        ],
+    ],
+)
+def test_connection_properties(dynamic_db, request, Constructor, alias, dialect):
+    dynamic_db = request.getfixturevalue(dynamic_db)
+
+    conn = Constructor(dynamic_db)
+
+    assert conn.alias == alias
+    assert conn.dialect == dialect
+
+
+@pytest.mark.parametrize(
+    "dynamic_db, Constructor, expected",
+    [
+        [
+            "setup_postgreSQL",
+            SQLAlchemyConnection,
+            "postgresql://ploomber_app:***@localhost:5432/db",
+        ],
+        [
+            "setup_duckDB",
+            SQLAlchemyConnection,
+            "duckdb:////tmp/db-duckdb",
+        ],
+        [
+            "setup_duckDB_native",
+            DBAPIConnection,
+            "DuckDBPyConnection",
+        ],
+        [
+            "setup_duckDB",
+            partial(SQLAlchemyConnection, alias="some-alias"),
+            "some-alias",
+        ],
+        [
+            "setup_duckDB_native",
+            partial(DBAPIConnection, alias="another-alias"),
+            "another-alias",
+        ],
+    ],
+)
+def test_connection_identifiers(
+    dynamic_db, request, monkeypatch, Constructor, expected
+):
+    dynamic_db = request.getfixturevalue(dynamic_db)
+
+    Constructor(dynamic_db)
+
+    assert len(ConnectionManager.connections) == 1
+    assert set(ConnectionManager.connections) == {expected}
+
+
+@pytest.mark.parametrize(
+    "dynamic_db, Constructor, expected",
+    [
+        [
+            "setup_postgreSQL",
+            SQLAlchemyConnection,
+            {
+                "dialect": "postgresql",
+                "driver": "psycopg2",
+                "server_version_info": ANY,
+            },
+        ],
+        [
+            "setup_duckDB",
+            SQLAlchemyConnection,
+            {
+                "dialect": "duckdb",
+                "driver": "duckdb_engine",
+                "server_version_info": ANY,
+            },
+        ],
+        [
+            "setup_duckDB_native",
+            DBAPIConnection,
+            {
+                "dialect": "duckdb",
+                "driver": "DuckDBPyConnection",
+                "server_version_info": ANY,
+            },
+        ],
+    ],
+    ids=[
+        "postgresql",
+        "duckdb",
+        "duckdb_native",
+    ],
+)
```

Comment:
Should we test more connections here? (mariadb, sqlite, snowflake, etc...)

Reply:
[fixed](https://github.com/ploomber/jupysql/pull/755/commits/bd5151872d824f4da1848165973b4d34bb72fd69)